File: spring-boot-kafka/src/main/java/org/chobit/spring/tools/DateKit.java
package org.chobit.spring.tools;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public abstract class DateKit {


    public static final String DEFAULT_PATTERN = "yyyy-MM-dd HH:mm:SS";


    public static Date parse(String date) throws ParseException {
        return new SimpleDateFormat(DEFAULT_PATTERN).parse(date);
    }


    public DateKit() {
        throw new UnsupportedOperationException("Private constructor, cannot be accessed.");
    }

}

File: scala-gradle-build/common/src/main/scala/chobit/common/DateKit.scala
package chobit.common

import java.text.SimpleDateFormat
import java.util.Date

object DateKit {

  val DEFAULT_PATTERN: String = "yyyy-MM-dd HH:mm:ss"

  def parse(source: String): Date = parse(DEFAULT_PATTERN, source)

  def parse(pattern: String, source: String): Date = new SimpleDateFormat(pattern).parse(source)

  def format(source: Date): String = format(DEFAULT_PATTERN, source)

  def format(pattern: String, timeInMills: Long): String = format(pattern, new Date(timeInMills))

  def format(pattern: String, source: Date): String = new SimpleDateFormat(pattern).format(source)

}
